tests/sessions_tests/tests.py

```diff
@@ -519,6 +519,20 @@ def test_non_default_cache(self):
         with self.assertRaises(InvalidCacheBackendError):
             self.backend()
 
+    def test_session_cache_failure(self):
+        max_value_length = 2**20
+
+        exceptions = []
+        key = "some key"
+        val = "a" * (max_value_length + 1)
+        try:
+            self.session[key] = val
+            self.session.save()
+        except Exception as e:
+            exceptions.append(e)
+
+        self.assertEqual(len(exceptions), 0)
+
```

Comment:
@sulabhkatila I have a failing test you can use that uses a deliberately failing cache

Add this to `tests/session_tests/cache.py`:

```python
from django.core.cache.backends.locmem import LocMemCache


class FailingCache(LocMemCache):
    def set(self, *args, **kwargs):
        raise Exception("Error saving to cache")
```

Then update your test like so:

```suggestion
    @override_settings(
        CACHES={"default": {"BACKEND": "sessions_tests.cache.FailingCache"}}
    )
    def test_session_cache_failure(self):
        self.session = self.backend()
        self.session.save()
```

Reply:
@shangxiao What's your view on defining this "failingcache" inside the `cache` tests, next to `closeable_cache.py` and `liberal_backend.py`? There's something that unsettles me about placing this file in this folder, but I can't quite pinpoint what it is that bothers me exactly.

Reply:
@nessita I have no particular view ☺️ Seems like a good place to put it 👍

Reply:
Amazing, thank you! Will do that.